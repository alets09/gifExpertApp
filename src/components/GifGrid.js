import React, { useState , useEffect} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {
    const {data, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeInDown">{category}</h3>
            {loading && <p className=" load animate__animated animate__zoomOutDown animate__delay-1s">Loading...</p>}
            <div className="card-grid">
                {
                    data.map(img => (
                        <GifGridItem key={img.id} {...img}/>
                    ))
                }
            </div>
        </> 
    )
}
