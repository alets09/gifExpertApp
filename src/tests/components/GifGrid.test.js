import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('pruebas para el componente GifGird', () => {
    const category = 'One punch';

    test('debera hacer match con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper = shallow( <GifGrid category={category} /> );   
        expect(wrapper).toMatchSnapshot(); 
    });
        
    test('debe de mostrar imagenes cuando se carga el useFechGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/cualquier.jpg',
            title: 'title cualquiera' 
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow( <GifGrid category={category} /> ); 
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
