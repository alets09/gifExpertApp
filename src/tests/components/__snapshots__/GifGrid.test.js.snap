// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas para el componente GifGird debe de mostrar imagenes cuando se carga el useFechGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="title cualquiera"
      url="http://localhost/cualquier.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas para el componente GifGird debera hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    One punch
  </h3>
  <p
    className=" load animate__animated animate__zoomOutDown animate__delay-1s"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
